core/amber/src/main/python/core/architecture/rpc/async_rpc_server.py

```diff
@@ -1,122 +1,124 @@
-from core.architecture.handlers.control.open_executor_handler import OpenExecutorHandler
-from core.architecture.handlers.control.pause_worker_handler import PauseWorkerHandler
-from core.architecture.handlers.control.query_current_input_tuple_handler import (
-    QueryCurrentInputTupleHandler,
-)
-from core.architecture.handlers.control.query_statistics_handler import (
-    QueryStatisticsHandler,
-)
-from core.architecture.handlers.control.replay_current_tuple_handler import (
-    ReplayCurrentTupleHandler,
-)
-from core.architecture.handlers.control.resume_worker_handler import ResumeWorkerHandler
-from core.architecture.handlers.control.start_worker_handler import StartWorkerHandler
-from core.architecture.handlers.control.add_input_channel_handler import (
-    AddInputChannelHandler,
-)
-from core.architecture.handlers.control.scheduler_time_slot_event_handler import (
-    SchedulerTimeSlotEventHandler,
-)
+import asyncio
 from core.architecture.managers.context import Context
+from core.architecture.rpc.async_rpc_handler_initializer import (
+    AsyncRPCHandlerInitializer,
+)
 from core.models.internal_queue import InternalQueue, ControlElement
 from core.util import get_one_of, set_one_of
-from proto.edu.uci.ics.amber.engine.architecture.worker import (
-    ControlCommandV2,
-    ControlException,
-    ControlReturnV2,
+from proto.edu.uci.ics.amber.engine.architecture.rpc import (
+    ReturnInvocation,
+    ControlRequest,
+    ControlInvocation,
+    ControlReturn,
+    ControlError,
+    ErrorLanguage,
 )
 from proto.edu.uci.ics.amber.engine.common import (
     ActorVirtualIdentity,
-    ControlInvocationV2,
     ControlPayloadV2,
-    ReturnInvocationV2,
 )
 
 
 class AsyncRPCServer:
     def __init__(self, output_queue: InternalQueue, context: Context):
-        self._context = context
         self._output_queue = output_queue
-        self._handlers: dict[type(ControlCommandV2), ControlHandler] = dict()
-        self.register(NoOpHandler())
-        self.register(StartWorkerHandler())
-        self.register(PauseWorkerHandler())
-        self.register(ResumeWorkerHandler())
-        self.register(OpenExecutorHandler())
-        self.register(AssignPortHandler())
-        self.register(AddPartitioningHandler())
-        self.register(AddInputChannelHandler())
-        self.register(QueryStatisticsHandler())
-        self.register(QueryCurrentInputTupleHandler())
-        self.register(InitializeExecutorHandler())
-        self.register(UpdateExecutorHandler())
-        self.register(ReplayCurrentTupleHandler())
-        self.register(EvaluateExpressionHandler())
-        self.register(SchedulerTimeSlotEventHandler())
-        self.register(WorkerDebugCommandHandler())
+        rpc_mapping = AsyncRPCHandlerInitializer(context).__mapping__()
+        self._handlers: dict[str, grpclib.const.Handler] = {
+            k.split("/")[-1].lower(): v for k, v in rpc_mapping.items()
+        }
```

Comment:
I kind of want to keep the old register logic as it can tell the IDE that one Handler is used somewhere. Otherwise all the handlers would seems to be unused. If there is another way to link their usage to the IDE, I am also fine with it.

Reply:
AsyncRPCHandlerInitializer now captures them.